watertap/property_models/activated_sludge/modified_asm2d_reactions.py

```diff
@@ -0,0 +1,102 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+"""
+ASM2d reaction package.
+
+Important Note: ASM2d reactions depend on the presences of TSS in solution, however
+TSS does not take part in the rate expressions. Thus, it is possible to have cases
+where there is insufficient TSS present in the system for the reactions to occur
+resulting in an infeasible solution.
+
+Reference:
+
+[1] Henze, M., Gujer, W., Mino, T., Matsuo, T., Wentzel, M.C., Marais, G.v.R.,
+Van Loosdrecht, M.C.M., "Activated Sludge Model No.2D, ASM2D", 1999,
+Wat. Sci. Tech. Vol. 39, No. 1, pp. 165-182
```

Comment:
Should you add a reference showing the introduction of K and Mg into ASM2D?

Reply:
The K and Mg are just added as two additional inert components and don't involve in any chemical precipitation process, so I don't have a good suggestion for reference of this modification.

Reply:
Yea, I agree that we don't need another reference here. Also, technically K and Mg aren't being added into this package at all - it's just the option to add any components that the user sees fit.

Reply:
Right now you're just adding components as inert species, but later on, it is possible that we might have precipitation reactions and or a charge balance. My point being that we might as well add the Flores-Alsina refs that consider modification of ASM2d/ADM1 in the first place.